features/graph/simple/src/main/java/org/opennms/netmgt/graph/simple/SimpleVertex.java

```diff
@@ -37,91 +37,79 @@
 import org.opennms.netmgt.graph.api.generic.GenericVertex;
 import org.opennms.netmgt.graph.api.info.NodeInfo;
 
-public class SimpleVertex implements Vertex, NodeAware, LocationAware {
+/**
+ * Acts as a domain specific view on a Vertex.
+ * Can be extended by domain specific Vertex classes.
+ * It contains no data of it's own but operates on the data of it's wrapped GenericVertex.
+ */
+public class SimpleVertex implements Vertex,
+        NodeAware, // TODO: patrick discuss with mvr if this is the right place for this interface
+        LocationAware // TODO: patrick discuss with mvr if this is the right place for this interface
```

Comment:
Yes and GenericVertex should also implement those interfaces.

Reply:
ok. I added them there as well